Add `VCPKG_FIXUP_MACHO_RPATH` (#346)
microsoft/vcpkg-docs: vcpkg/users/triplets.md

[CMAKE_INSTALL_NAME_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_NAME_DIR.html)
for more information.

### VCPKG_FIXUP_MACHO_RPATH

> Ensures Mach-O binaries built by vcpkg are relocatable, i.e. with relative install names and run-paths.

Reviewer: ```suggestion
Ensures Mach-O binaries built by vcpkg are relocatable by using relative install names and run-paths.
```

When set to `ON`:
- Modifies absolute `LC_LC_ID_DYLIB` field to `@rpath/<library>` for shared library binaries;

> - Modifies absolute `LC_RPATH` fields to relative `@loader_path/<relative/path/to/library>` for executable and shared library binaries.

Reviewer: It would be nice to have external links to documentation about these.

> [!NOTE]  

> > This functionality is implicitly enabled by default when `VCPKG_TARGET_IS_OSX` evaluates to `TRUE`. May be disabled by explicitly setting to `OFF`.

Reviewer: ```suggestion
> This functionality is enabled by default when `VCPKG_TARGET_IS_OSX` is `TRUE`. To disable, explicitly set `VCPKG_FIXUP_MACHO_RPATH` to `OFF` in a [triplet file](../concepts/triplets.md).
```
